source/includes/_feature-extending.md

```diff
@@ -82,4 +84,81 @@ HTTP/1.1 202 Accepted
 
 ```
 
-Personal variables will be ignored and not copied.
\ No newline at end of file
+Personal variables will be ignored and not copied.
+
+### Merging a filtered dataset
+
+If a filter is necessary on the right dataset, it is possible to indicate it
+on the `adapt` expression so that only the selected rows will be brought over
+during the join. This can work in conjunction with selecting a subset of
+variables as well.
+
+To do so, it is necessary to add a `filter` attribute on the payload that can
+contain either a filter expression, wrapped under `{"expression": <expr>}` of
+an existing filter URL wrapped as `{"filter": <url>}` - Note that the filter
+must exist on the right dataset for its expression to match.
+
+Filtering with an expression:
+
+```http
+POST /datasets/{dataset_id}/variables/ HTTP/1.1
+Content-Type: application/json
+
+{
+    "function": "adapt",
+    "args": [{
+        "dataset": "/datasets/{other_id}/"
+    }, {
+        "variable": "/datasets/{other_id}/variables/{right_key_id}/"
+    }, {
+        "variable": "/datasets/{dataset_id}/variables/{left_key_id}/"
+    }],
+    "filter": {
+        "expression": {
```

Comment:
I'm not aware of anywhere else where we require this "expression" nesting for filters. It's always straight to "function": "==", .... Remove it here to match our existing API convention.